runtime/vm/ContinuationHelpers.cpp

```diff
@@ -1238,6 +1238,24 @@ takeVirtualThreadListToUnblock(J9VMThread *currentThread)
 				/* Skip vthreads that are still in transition. */
 				switch (state) {
 				case JAVA_LANG_VIRTUALTHREAD_BLOCKING:
+				{
+					omrthread_monitor_t monitor = syncObjectMonitor->monitor;
+					/* For BLOCKING and BLOCKED states VirtualThread.blockPermit will allow
+					 * VirtualThread.afterYield() to reschedule the thread.
+					 * Only set blockPermit if the monitor blocker is available.
+					 *
+					 * All blocking/waiting monitor have to be inflated, if the monitor has not been inflated,
+					 * then the owner have not yet released the flatlock.
+					 */
+					if (J9_ARE_ANY_BITS_SET(((J9ThreadMonitor *)monitor)->flags, J9THREAD_MONITOR_INFLATED)) {
+						if (0 == monitor->count) {
+							J9VMJAVALANGVIRTUALTHREAD_SET_BLOCKPERMIT(currentThread, current->vthread, JNI_TRUE);
+						}
+					}
```

Comment:
This can be reduced to a single `if` statement.
```suggestion
					if (J9_ARE_ANY_BITS_SET(((J9ThreadMonitor *)monitor)->flags, J9THREAD_MONITOR_INFLATED)
					&& (0 == monitor->count)
					) {
						J9VMJAVALANGVIRTUALTHREAD_SET_BLOCKPERMIT(currentThread, current->vthread, JNI_TRUE);
					}
```